music/src/core/player.ts

```diff
@@ -541,3 +541,134 @@ export class PlayerWithClick extends Player {
     super(true, callbackObject);
   }
 }
+
+/**
+ * A `NoteSequence` player that uses a MIDI output for playing. Note that
+ * WebMIDI is not supported in all browsers. A
+ * [polyfill](https://cwilso.github.io/WebMIDIAPIShim/) exists, but it too
+ * requires a plugin to be installed on the user's computer, so it might not
+ * work in all cases.
+ *
+ * If you want to use a particular MIDI output, you must update the `output`
+ * property before calling `start`, otherwise a message will be sent to
+ * all connected MIDI outputs:
+ *
+ * Example (easy mode):
+ *
+ *   `
+ *    const player = new mm.MIDIPlayer();
+ *    player.requestMIDIAccess().then(() => {
+ *      player.output = [player.midiOutputs[0]];
+ *      player.start();
+ *    })`
+ *
+ * You can also explicitly request MIDI access outside of the player, in
+ * your application, and just update the `output` property before playing:
+ *
+ * Example (advanced mode):
+ *
+ *   `
+ *    navigator.requestMIDIAccess().then((midi) => {
+ *       // Get whichever MIDI outputs you want.
+ *       const outputs = initOutputs(midi);
```

Comment:
what is initOutputs?

Reply:
some code that you would write to initialize all the midi outputs. should i write it all out you think?

Reply:
What does it mean to "initialize" the outputs?

Reply:
like...the bit that does `navigator.requestMidiAccess()` and then iterates through all the midiAccess.outputs to get the values

Reply:
isn't navigator.requestMidiAccess already being called?

Reply:
the bit where you loop through outputs. i'll make this explicit.